File: 2451. Odd String Difference/solution.py
class Solution(object):
  def oddString(self, words):
    """
    :type words: List[str]
    :rtype: str
    """
    dictionary = {}
    for word in words:
      calc = []
      for index in range(1, len(word)):
        difference = ord(word[index]) - ord(word[index - 1])
        calc.append(difference)
      stringCalc = str(calc)
      if stringCalc not in dictionary:
        dictionary[stringCalc] = [word]
      else:
        dictionary[stringCalc].append(word)
    
    for difference in dictionary:
      if len(dictionary[difference]) == 1:
        return dictionary[difference][0]

File: 2451. Odd String Difference/solution.js
/**
 * @param {string[]} words
 * @return {string}
 */
 var oddString = function(words) {
  let dictionary = {};
  for (let index = 0; index < words.length; index++){
    let word = words[index];
    let calc = [];
    for (let index2 = 1; index2 < word.length; index2++){
      let difference = word[index2].charCodeAt(0) - word[index2 - 1].charCodeAt(0);
      calc.push(difference);
    }

    if (dictionary[calc]){
      dictionary[calc].push(word);
    }
    else {
      dictionary[calc] = [word];
    }
  }

  for (let difference in dictionary){
    if ((dictionary[difference]).length == 1){
      return dictionary[difference][0];
    }
  }
};